src/contracts/Repository/Events/Notification/BeforeMarkNotificationAsUnreadEvent.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Core\Repository\Events\Notification;
+
+use Ibexa\Contracts\Core\Repository\Event\BeforeEvent;
+use Ibexa\Contracts\Core\Repository\Values\Notification\Notification;
+
+final class BeforeMarkNotificationAsUnreadEvent extends BeforeEvent
+{
+    /** @var \Ibexa\Contracts\Core\Repository\Values\Notification\Notification */
+    private $notification;
```

Comment:
```suggestion
    private Notification $notification;
```

This is new code for a 4.6 so you can use all PHP 7.4 features